dan-stineback/test/test.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const fs = require('fs');
+const expect = require('chai').expect;
+const jsonParser = require(__dirname + '/../lib/jsonParser');
+const chai = require('chai')
+const chaiHTTP = require('chai-http');
+chai.use(chaiHTTP);
+const request = chai.request;
+
+
+
+describe('json parser test', () => {
+  it('should read valid json', (done)=>{
+    let req = fs.createReadStream(__dirname + '/good.json');
+    jsonParser(req, null, () =>{
+      expect(req.body.message).to.eql('test');
+      done();
+    });
+  });
+  it('should reject invalid json', (done)=>{
+    let req = fs.createReadStream(__dirname + '/bad.json');
+    jsonParser(req, null, (err) =>{
+      expect(err.message).to.eql('invalid JSON');
+      expect(err.statusCode).to.eql(422);
+      done();
+  });
+});
+});
+
+describe('should run integration tests', ()=>{
+  it('should read good JSON', (done)=>{
+   require(__dirname + '/../lib/server');
+   request('localhost:3000')
+      .post('/zoots')
+       .send({message:'test2'})
+       .end((err, res)=>{
+         expect(err).to.eql(null);
+         expect(res.text).to.eql('JSON parsed')
+         done();
+       });
+   });
+   it('should read bad JSON', (done)=>{
+     require(__dirname + '/../lib/server');
+     request('localhost:3000')
```

Comment:
This should go in a mocha before and not at this point.  This will guarantee that the server is up when you go to make the request.  I didn't notice this at first because tests were passing, but if you're server was loading a lot of middleware it could take longer and your request would not find the server.  Ideally you would close the server in a mocha after. Not points off because we haven't explicitly pointed this out before.